tiny_dnn/core/framework/tensor.h

```diff
@@ -264,83 +405,61 @@ class Tensor {
     }
 
     void *mutable_device_data() {
+        static_assert(!kConst, "Non-constant operation on constant Tensor");
         toDevice();
         data_dirty_ = true;
         return (*device_data_)();
     }
 #endif
 
     size_t size() const {
-        return host_data_.size();
+        return calcSize();
```

Comment:
why not `storage_ptr_->size()` ? To avoid calculating each time
Where:
```cpp
size_t TensorStorage::size() const {
    return host_data_.size();
}
```